usaspending_api/common/logging.py

```diff
@@ -204,3 +222,90 @@ def ensure_logging(
             f"logger={cfg_logger_name}, formatter={cfg_formatter_name}, formatter_class={str(formatter_class)})"
         )
     return logger_to_use
+
+
+# CUSTOM Logging EXPORTER for debugging
+class LoggingSpanProcessor(SpanProcessor):
+    def __init__(self):
+        self._lock = threading.Lock()
+        self._flushed = True  # Simulates the state of flushing
+
+    def on_end(self, span: ReadableSpan) -> None:
+        trace_id = span.context.trace_id
+        span_id = span.context.span_id
+        logger.debug(f"Span ended: trace_id={trace_id}, span_id={span_id}, {span.name}_attributes={span.attributes}")
+
+    def force_flush(self, timeout_millis: int = 30000) -> bool:
+        """Simulates flushing all spans within the given timeout."""
+        start_time = time_ns()
+        with self._lock:
+            # Simulate some processing delay for flushing
+            while not self._flushed:
+                elapsed_time = (time_ns() - start_time) / 1_000_000  # Convert to milliseconds
+                if elapsed_time > timeout_millis:
+                    logger.warning("force_flush timed out.")
+                    return False
+                # Simulate work by sleeping briefly
+                time.sleep(0.01)
+
+        logger.debug("All spans flushed successfully.")
+        return True
+
+
+class CustomAttributeSpanProcessor(SpanProcessor):
+    def __init__(self, attribute_key, attribute_value):
+        self.attribute_key = attribute_key
+        self.attribute_value = attribute_value
+
+    def on_start(self, span, parent_context):
+        # Add the custom attribute when the span starts
+        span.set_attribute(self.attribute_key, self.attribute_value)
+
+
+def add_custom_attribute_span_processors(tracerProvider: TracerProvider, attribute_pairs: List[Tuple[str, str]]):
+    for key, value in attribute_pairs:
+        custom_attribute_span_processor = CustomAttributeSpanProcessor(key, value)
+        tracerProvider.add_span_processor(custom_attribute_span_processor)
+
+
+def configure_logging(service_name="usaspending-api"):
+    # Set up the OpenTelemetry tracer provider
+    resource = Resource.create(attributes={"service.name": service_name})
+    provider = TracerProvider(resource=resource)
+    trace.set_tracer_provider(provider)
+
+    # Modify the following to add/remove information inside traces
+    # The following will be added to every trace
+    attribute_pairs = [
+        ("DEFUALT_DB_HOST", DATABASES["default"]["HOST"]),
+        ("TRACE_ENV", TRACE_ENV),
+    ]
+
+    add_custom_attribute_span_processors(trace.get_tracer_provider(), attribute_pairs)
+
+    # Set up the OTLP exporter
+    # Check out https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/
+    exporter = None
+    if os.getenv("USASPENDING_DB_HOST") == "127.0.0.1" and os.getenv("TOGGLE_OTEL_CONSOLE_LOGGING") == "True":
```

Comment:
Could we use the `settings.IS_LOCAL` instead? Because the host could be `localhost`, `127.0.0.1`, a container name, or even `0.0.0.0`.

Reply:
Yes, I think this makes sense 👍